error-prone-contrib/src/main/java/tech/picnic/errorprone/refasterrules/FileRules.java

```diff
@@ -40,4 +43,22 @@ String after(Path path) throws IOException {
       return Files.readString(path);
     }
   }
+
+  /**
+   * Prefer {@link Files#createTempFile(String, String, FileAttribute[])} over alternatives that
+   * create files will more liberal permissions.
+   */
+  static final class FilesCreateTempFileToFile {
+    @BeforeTemplate
+    @SuppressWarnings("java:S5443" /* This violation will be rewritten. */)
+    File before(String prefix, String suffix) throws IOException {
+      return Refaster.anyOf(
+          File.createTempFile(prefix, suffix), File.createTempFile(prefix, suffix, null));
+    }
+
+    @AfterTemplate
+    File after(String prefix, String suffix) throws IOException {
+      return Files.createTempFile(prefix, suffix).toFile();
```

Comment:
SonarCloud now compains about this line, while the [documentation](https://rules.sonarsource.com/java/type/Security Hotspot/RSPEC-5443) literally says:
```java
File f = Files.createTempFile("prefix", "suffix").toFile();  // Compliant
```

I suppose this is something to be reported upstream (either it's incorrect, or unclear).

Reply:
Alright, I had a look, and for POSIX file systems at least this is safe: [(a)](https://github.com/openjdk/jdk/blob/367e0a65561f95aad61b40930d5f46843fee3444/src/java.base/share/classes/java/nio/file/TempFileHelper.java#L91-L117), [(b)](https://github.com/openjdk/jdk/blob/367e0a65561f95aad61b40930d5f46843fee3444/src/java.base/share/classes/java/nio/file/TempFileHelper.java#L66-L67). I tested this on my Linux laptop: the suggested code produces a file with `-rw-------` permissions, while the other two variants create a file with `-rw-rw-r--` permissions.

The question is then what happens on Windows, but regardless this change is an improvement. I've added a comment to this effect, and suggest that this PR is good to go.